packages/interactivity/src/directives.js

```diff
@@ -330,7 +330,9 @@ export default () => {
 	// data-wp-text
 	directive( 'text', ( { directives: { text }, element, evaluate } ) => {
 		const entry = text.find( ( { suffix } ) => suffix === 'default' );
-		element.props.children = evaluate( entry );
+		const evaluatedEntry = evaluate( entry );
+		if ( typeof evaluatedEntry !== 'string' ) return null;
+		element.props.children = evaluatedEntry;
```

Comment:
> PD: Question from @DAreRodz: should we allow numbers?

Oh, absolutely.

However, I'm wondering if we should leverage `.toString()` instead:

```suggestion
    try {
      element.props.children = evaluate( entry ).toString();
    } catch (e) {
      element.props.children = null;
    }
```

For debugging purposes it'd be nicer because it can print booleans, numbers, even arrays or functions.

What do you think?

Reply:
Yup, that implementation would be great. 👍  We can get rid of potential `undefined` or `null` strings as those values don't have a `.toString()` function, and numbers, booleans, or even arrays would be printed nicely.

My only concern would be about the possibility of having `[object Object]` too often, but I don't think it's that much of a problem. 😄

Reply:
Done in [e06d980](https://github.com/WordPress/gutenberg/pull/57879/commits/e06d980796ee8e7c284807c91ae79e51bf771ec1)

Reply:
> My only concern would be about the possibility of having [object Object] too often, but I don't think it's that much of a problem. 😄

We are indeed having them for components.
![Screenshot 2024-01-16 at 13 46 06](https://github.com/WordPress/gutenberg/assets/37012961/2758ae45-50e9-49b2-8358-df6a5df9a15f)